### ISSUE TITLE
NE site specific water use data

### ISSUE THREAD

**Issue by amabdallah** [WSWCWaterDataExchange/MappingStatesDataToWaDE2.0#39]
**Homepage**
It would be best if we can get a shapefile for all the sites 
https://nednr.nebraska.gov/RealTime/

**List**
https://nednr.nebraska.gov/RealTime/Stations/StationsList

A good start is to create the site, beneficial use, variable, source tables in WaDE. 

Not sure if there is an API endpoint to get the archived-time series data. 
"Currently data is available for the past two years, in one year intervals"

**Comment by amabdallah** [comment]
This site seems to offer water use data, but it's protected. Need to check with NDR 

https://nednr.nebraska.gov/wateruse

**Comment by rwjam** [comment]
> Not sure if there is an API endpoint to get the archived-time series data.
> "Currently data is available for the past two years, in one year intervals"

Did we ever hear if there was an API endpoint to get all archived-time series data?  Current approach would involve manually selecting each site, and downloading data from each 2-3 time series related to that plot site.  Very cumbersome.

**Comment by rwjam** [comment]
NE data hub found [here](https://www.nebraskamap.gov/).  Stream gauge location data available at **Inland Water** - > **Stream Gages**.  However, there is no time series amount information tied to location data.

**Comment by amabdallah** [comment]
You're right, this map does not provide data. I looked further and was able to find this other page that has download options (like  CSV). We're only interested in stations that are run by NE, I clicked at a few URLs of those stations and they return an html page with errors. I'm guessing they updated the system and these pages still point to the old one. I'll ask them.
https://www.nebraskamap.gov/datasets/nebraska::stream-gages/about

![image](https://user-images.githubusercontent.com/3268971/123869960-e6260700-d8ee-11eb-891d-553679056ec7.png)

**Comment by amabdallah** [comment]
See info under the **SWAGGER API** header.
I was told by our Nebraska POC during the workshop in SLC that this API is up and running now, and the API call GetStationList has the coordinates which were not available before.
https://dnr.nebraska.gov/improved-access-streamgaging-information

After getting info about the stations, I think we want this 
[GET](https://nednr.nebraska.gov/IwipApi/swagger/ui/index#!/StreamGage/StreamGage_DailyMeanByYear) [/api/v1/StreamGage/DailyMeanByYear](https://nednr.nebraska.gov/IwipApi/swagger/ui/index#!/StreamGage/StreamGage_DailyMeanByYear)

**Comment by rwjam** [comment]
Digging through NE reservoir gage data
- Looks like their data on both the stream gage shapefile ([link](https://www.nebraskamap.gov/datasets/nebraska::stream-gages/explore?location=41.324351%2C-100.320124%2C7.91)) and the swagger hub API page ([link](https://nednr.nebraska.gov/IwipApi/swagger/ui/index#/StreamGage)) don't line up perfectly.  The shapefile says the site is active, but then the API returns only empty strings.  Likewise the API will say the site is active, but there is no record of it in the shapefile.
- Possible that shapefile isn't being updated often enough.
- Think I'll just try pulling sites from the GET/api/v1/StreamGage/GetStationList service, then try to match that GEt/api/v1/StreamGage/DailyMeanByYear service.

**Comment by amabdallah** [comment]
Nice work. Sure go ahead and try to get as much data as possible and we'll ask the state to review or help with issues.

**Comment by rwjam** [comment]
Interesting enough, the results from each of the StreamGage services on their swagger hub API page ([link](https://www.nebraskamap.gov/datasets/nebraska::stream-gages/explore?location=41.324351%2C-100.320124%2C7.91)) don't seem to line up.
- DailyMeanByYear returns a daily mean by **DAY** value, not an aggregated year value.
- DailyMeanByMonth might be rounding the value too much as the aggregation in the ByMonth services dones't seem to match that returend by the ByYear service.
- DailyMeanByDay tends to return nothing.

Will probably just use the DailyMeanByYear service, but mark it down in our system as a Daily value.

**Comment by amabdallah** [comment]
Great review. I'll reach out to our POC and let them know. It's a new service, so I hope they can make fixes fast. 
In the meantime, yes, let's get the sites and daily 2022 values in our system

**Comment by rwjam** [comment]
Uploaded.  Unsure of time frame of available year.  Metadata not available.  Used range of 2000-2022 as temp fix.

**Comment by rwjam** [comment]
Few noted issues.

- We are getting both sites and timeseries from this [api service (link),](https://nednr.nebraska.gov/IwipApi/swagger/ui/index#/StreamGage) using the GetStationList service and the DailyMeanByYear service.  As of the moment the service is only returning two sites with a SiteTypeCV = reservoir, but we are only able to match timeseries info to one of them.  We are using a limited set time series search from 2000-2022 as we had trouble working with the api service as it was defaulting only the most current year’s values.  We could try expanding the year search but I’m not sure that will help.
- Example return timeseries return for the Willow Creek Reservoir near Pierce ([link](https://nednr.nebraska.gov/IwipApi/api/v1/StreamGage/DailyMeanByYear?StationNumber=00410300&MeanYear=2010)).
- For NE I’m not seeing an indication if the flow data at a SiteTypeCV = reservoir is intake or outtake.
